skills/repo_test.go

```diff
@@ -740,3 +740,79 @@ func (s *RepositorySuite) TestGrantsOfReturnsCorrectData(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(s.testRepo.GrantsOf(s.skill.Snap, s.skill.Name), HasLen, 0)
 }
+
+// Tests for Repository.SecuritySnippetsForSnap()
+
+func (s *RepositorySuite) TestSlotSnippetsForSnapSuccess(c *C) {
+	var testSecurity SecuritySystem = "security"
+	t := &TestType{
+		TypeName: "type",
+		SkillSecuritySnippetCallback: func(skill *Skill, securitySystem SecuritySystem) ([]byte, error) {
+			if securitySystem == testSecurity {
+				return []byte(`producer snippet`), nil
+			}
+			return nil, ErrUnknownSecurity
+		},
+		SlotSecuritySnippetCallback: func(skill *Skill, securitySystem SecuritySystem) ([]byte, error) {
+			if securitySystem == testSecurity {
+				return []byte(`consumer snippet`), nil
+			}
+			return nil, ErrUnknownSecurity
+		},
+	}
+	repo := s.emptyRepo
+	err := repo.AddType(t)
+	c.Assert(err, IsNil)
```

Comment:
Maybe `c.Assert(repo.AddType(t), IsNil)` would make this test read more concisely